### ISSUE TITLE
Replace with open source alternative?

### ISSUE THREAD

**Issue by pombredanne** [rq/rq#2062]
Redis is no longer open source going forward....

https://redis.com/blog/redis-adopts-dual-source-available-licensing/

> Redis will no longer be distributed under the three-clause Berkeley Software Distribution (BSD).

**Comment by selwin** [comment]
Yeah, this sucks. But let's see how this plays out.

**Comment by ankush** [comment]
SSPLv1 while not as permissive still allows full commercial use as long as you also open source your modification of redis and code that runs the managed service you provide. 

The license change doesn't affect vast majority of users use Redis for RQ in their application, as long as they are not offering it as a service. Anyway, IANAL so confirm with someone who is before doing anything. 


ref: https://www.mongodb.com/legal/licensing/server-side-public-license/faq

**Comment by SpecLad** [comment]
FWIW, the client libraries remain open source.

**Comment by pombredanne** [comment]
@ankush SSPL is a source-available, proprietary license. I cannot deploy my open source projects based on proprietary code, as I do not know what my users will use it for, commercial or not, and I do not want them to have to worry about this. And though it may be OK for you to use it internally, I would think twice as the next license ratchet step from Redis will likely to make more proprietary, and eventually fully commercial-only. 

Why? The writing is on the wall :) .... in 2018, Redis exec said:
https://redis.io/blog/redis-license-bsd-will-remain-bsd/

> We want to address your questions and be crystal clear: the license for open source Redis was never changed. It is BSD and will always remain BSD.

... and it is crystal clear now that this is a promise they obviously did not keep. So I would not trust them in the long run.

@selwin There are a few decent Redis alternatives that are emerging:

- https://github.com/apache/kvrocks (compatible, Apache-licensed)
- https://github.com/valkey-io/valkey (a fork of Redis, BSD-licensed)

A possible way could be to start testing if these can work as well as redis? Or just wait and see as eventually Linux distros will likely start to deploy these too as the SSPL-licensed Redis versions will no longer be part of their official packages.

( @selwin unrelated, you profile-listed home at http://ong.co.id looks inactive? )

**Comment by selwin** [comment]
> A possible way could be to start testing if these can work as well as redis?

It would be great if someone can experiment with this.

For now, I want to focus on releasing a last stable 1.16.x version before putting my effort into releasing 2.0 which should take a few more months.

In the medium to long run, I definitely want to support an open source Redis alternative. I personally think Valkey would win because it has the support of Linux Foundation, but I want to wait for a few months before deciding on which Redis alternative to support.

The ecosystem around databases typically take time to mature and even Valkey still don't have the question around [redis-py](https://github.com/valkey-io/valkey/issues/164) figured out.

**Comment by lambda-science** [comment]
https://github.com/rq/rq/pull/2223
Does this means that Valkey is supported ?

**Comment by selwin** [comment]
@lambda-science yes, the latest RQ release officially supports Valkey. This was merged in https://github.com/rq/rq/pull/2223
